lua/acf/core/utilities/util_sv.lua

```diff
@@ -399,97 +399,150 @@ do -- Entity linking
-			end
+		local LinkData2 = ACF.GetLinkDataSafe(Class1,Class2)
+		LinkData2.Link = function(Ent2, Ent1, FromChip)
+			if LinkData2.LinkStore1 then AddLink(Ent1, Ent2, LinkData2.LinkStore1) end
+			if LinkData1.LinkStore2 then AddLink(Ent1, Ent2, LinkData1.LinkStore2) end
+			return Function(Ent1, Ent2, FromChip)
 		end
 	end
 
-	--- Registers that two classes can be linked, as well as how to handle entities of their class being linked.
+	--- Registers that two classes can be unlinked, as well as how to handle entities of their class being unlinked.
 	--- @param Class1 string The first class in the link
 	--- @param Class2 string The other class in the link
-	--- @param Function fun(Entity1:table, Entity2:table) The linking function defined between an entity of Class1 and an entity of Class2; this should always return a boolean for link status and a string for link message
-	function ACF.RegisterClassLink(Class1, Class2, Function)
-		RegisterNewLink("Link", Class1, Class2, Function)
+	--- @param Function fun(Entity1:table, Entity2:table, FromChip:boolean) The unlinking function defined between an entity of Class1 and an entity of Class2
+	function ACF.RegisterClassUnlink(Class1, Class2, Function)
+		local LinkData1 = ACF.GetLinkDataSafe(Class1,Class2)
+		LinkData1.Unlink = function(Ent1, Ent2, FromChip)
+			if LinkData1.LinkStore1 then RemoveLink(Ent1, Ent2, LinkData1.LinkStore1) end
+			if LinkData1.LinkStore2 then RemoveLink(Ent1, Ent2, LinkData1.LinkStore2) end
+			return Function(Ent1, Ent2, FromChip)
+		end
+		if Class1 == Class2 then return end
+		local LinkData2 = ACF.GetLinkDataSafe(Class1,Class2)
+		LinkData2.Unlink = function(Ent2, Ent1, FromChip)
+			if LinkData2.LinkStore1 then RemoveLink(Ent1, Ent2, LinkData2.LinkStore1) end
+			if LinkData1.LinkStore2 then RemoveLink(Ent1, Ent2, LinkData1.LinkStore2) end
+			return Function(Ent1, Ent2, FromChip)
+		end
 	end
 
-	--- Returns the callback defined previously by ACF.RegisterClassLink between Class1 and Class2.
+	--- Registers a validation check between two classes.
 	--- @param Class1 string The first class in the link
 	--- @param Class2 string The other class in the link
-	--- @return fun(Entity1:table, Entity2:table) | nil # The linking function defined between an entity of Class1 and an entity of Class2, or nil if Class1 has no linking functions
-	function ACF.GetClassLink(Class1, Class2)
-		if not ClassLink.Link[Class1] then return end
-
-		return ClassLink.Link[Class1][Class2]
+	--- @param Function fun(Entity1:table, Entity2:table, FromChip:boolean) The checking function defined between an entity of Class1 and an entity of Class2
+	function ACF.RegisterClassCheck(Class1, Class2, Function)
+		ACF.GetLinkDataSafe(Class1,Class2).Check = function(Ent1, Ent2, FromChip) return Function(Ent1, Ent2, FromChip) end
+		if Class1 == Class2 then return end
+		ACF.GetLinkDataSafe(Class2,Class1).Check = function(Ent2, Ent1, FromChip) return Function(Ent1, Ent2, FromChip) end
```

Comment:
Readability, localize the result from `ACF.GetLinkDataSafe` and then assign it's `Check` value.